In [215]:
import pandas as pd
import collections
from utils import *

In [2]:
OUTDIR = 'output'
INDIR_BILL = './data/bills' 
INDIR_SUMMARY = './data/summary' 

In [ ]:
# version_def = pd.read_csv('version_definition.csv')
# version_def['Version Abbreviation'] = version_def['Version Abbreviation'].str.upper()

In [199]:
# bills = []

# for root, dirs, filenames in os.walk(INDIR_BILL):
#     for filename in filenames:
#         b_type, b_subtype, b_number, b_version, b_filename = extract_nameinfo(filename)
        
#         with open(os.path.join(INDIR_BILL, filename), 'rb') as f:
#             doc = xmltodict.parse(f.read())
            
#             b_tags = [key for key, value in recursive_items(doc)]
#             b_tags = list(set(b_tags))
            
#             b_dict = {'Type': b_type, 
#                       'Subtype': b_subtype, 
#                       'Number': b_number,
#                       'Version Abbreviation': b_version,
#                       'Filename': b_filename,
#                       'Tags': b_tags
#                      }
            
#             bills.append(b_dict)
            
# df_bills = pd.DataFrame(bills)
# df_bills = pd.merge(df_bills, version_def, how='left', on='Version Abbreviation', suffixes=('', '_'))
# df_bills.to_csv('tags.csv', index=False)

# Bills

In [202]:
df_bills.read_csv('tags.csv', index=False, header=True)

In [211]:
df_bills.groupby(['Type']).count()[['Filename']].rename(columns={'Filename': 'Count'})

Count
Type                         
Bills                    6726
Concurrent Resolutions    147
Joint Resolutions         241
Simple Resolutions        924

* 15 of 6726 bills are amendment document. (`EAH` and `EAS`)

In [206]:
df_bills.groupby(['Type', 'Subtype', 'Version Abbreviation', 'Version']).count()[['Filename']].rename(columns={'Filename': 'Count'})

Count
Type                   Subtype Version Abbreviation Version                            
Bills                  HR      EAH                  Engrossed Amendment House         2
                               EAS                  Engrossed Amendment Senate       10
                               EH                   Engrossed in House              300
                               ENR                  Enrolled Bill                    28
                               IH                   Introduced in House            3743
                               PCS                  Placed on Calendar Senate        18
                               RCS                  Reference Change Senate           1
                               RDS                  Received in Senate               28
                               RFS                  Referred in Senate              256
                               RH                   Reported in House               229
                               RS                   Reported in Senate               24
                       S       CPS                  Considered and Passed Senate      2
                               EAH                  Engrossed Amendment House         3
                               ENR                  Enrolled Bill                    11
                               ES                   Engrossed in Senate              71
                               IS                   Introduced in Senate           1787
                               PCS                  Placed on Calendar Senate        15
                               RCS                  Reference Change Senate           2
                               RFH                  Referred in House                40
                               RH                   Reported in House                 1
                               RS                   Reported in Senate              155
Concurrent Resolutions HCONRES EH                   Engrossed in House               10
                               ENR                  Enrolled Bill                     6
                               IH                   Introduced in House              73
                               RDS                  Received in Senate                8
                               RFS                  Referred in Senate                2
                               RH                   Reported in House                 4
                       SCONRES ATS                  Agreed to Senate                  5
                               ENR                  Enrolled Bill                     6
                               ES                   Engrossed in Senate               7
                               IS                   Introduced in Senate             18
                               PCS                  Placed on Calendar Senate         1
                               RFH                  Referred in House                 2
                               RS                   Reported in Senate                1
                               SC                   Sponsor Change                    4
Joint Resolutions      HJRES   EH                   Engrossed in House               19
                               ENR                  Enrolled Bill                    15
                               IH                   Introduced in House             116
                               PCS                  Placed on Calendar Senate         1
                               RDS                  Received in Senate               17
                               RFS                  Referred in Senate                2
                               RH                   Reported in House                 2
                       SJRES   CPS                  Considered and Passed Senate      1
                               ENR                  Enrolled Bill                     6
                               ES 

### Amendment

In [209]:
df_bills[df_bills['Version Abbreviation'].isin(['EAS', 'EAH'])].groupby(['Type', 'Subtype', 'Version Abbreviation', 'Version']).count()[['Filename']].rename(columns={'Filename': 'Count'})

Count
Type  Subtype Version Abbreviation Version                          
Bills HR      EAH                  Engrossed Amendment House       2
              EAS                  Engrossed Amendment Senate     10
      S       EAH                  Engrossed Amendment House       3

### 

In [210]:
df_bills[df_bills['Type'].isnull()]

Filename Number Subtype  \
519  92510.140382952224512.47152.xml    NaN     NaN   

                                                  Tags Type  \
519  [dc:publisher, paragraph, external-xref, form,...  NaN   

    Version Abbreviation Version Definition Chambers  
519                  NaN     NaN        NaN      NaN